### PR TITLE
Logic: Refactor `planner_list_all` to `planner_list`

### PR DESCRIPTION
**Pull request by Letm3through**
After the discussion, we decided to refactor original 'planner_list' to 'planner_show'.
  
In order to be consistent with 'list' command, let's refactor the current 'planner_list_all' command to 'planner_list'.

This consistency will allow a user to easily perceive the command usage and less likely to cause the confusion when trying to figure out the command for listing the degree plan.

In addition, let's update 'planner_list' to utilize 'DegreePlanner:toString' method instead of current method using 'StringBuilder' since the format for 'DegreePlanner:toString' was refined to print each 'DegreePlanner' nicely.

Moreover, along with the update for test cases regarding refactored and updated `planner_list`, some methods in 'ModelHelper' and 'ApplicationSystemTest' to allow other 'DegreePlanner' related system tests to use have been added.

The followings are the commits made:
* [1/2] [Logic: Refactor 'planner_list_all' to 'planner_list'](https://github.com/CS2113-AY1819S2-T09-1/main/pull/208/commits/559ad5ed0b10faa9fa0eea5b2d077963ee9abd49)
* [2/2] [Test: update for 'planner_list'](https://github.com/CS2113-AY1819S2-T09-1/main/pull/208/commits/18022f07f97dcf066bd1c9a73a1facdd0ad6fdf2)

### CODE REVIEW

File: src/test/java/pwe/planner/logic/LogicManagerTest.java [hunk 1]
```diff
@@ -86,22 +85,13 @@ public void execute_validCommand_success() {
...
-                degreePlannerListContent.append("No module inside");
-            } else {
-                degreePlannerListContent
-                        .append("Modules: " + degreePlanner.getCodes().stream().map(Code::toString).collect(
-                                Collectors.joining(", ")));
-            }
-            degreePlannerListContent.append("\n\n");
-        }
+        String plannerListCommand = PlannerListCommand.COMMAND_WORD;
+        String degreePlannerListContent =
```
Comment (chyeo) [inline src/test/java/pwe/planner/logic/LogicManagerTest.java]: You can compress the following lines into 2 lines

File: src/test/java/pwe/planner/logic/commands/PlannerListCommandTest.java [hunk 2]
```diff
@@ -15,46 +15,33 @@
...
-            if (degreePlanner.getCodes().isEmpty()) {
-                degreePlannerListContent.append("No module inside");
-            } else {
-                degreePlannerListContent
-                        .append("Modules: " + degreePlanner.getCodes().stream().map(Code::toString).collect(
-                                Collectors.joining(", ")));
-            }
-            degreePlannerListContent.append("\n\n");
-        }
+        String degreePlannerListContent =
```
Comment (chyeo) [inline src/test/java/pwe/planner/logic/commands/PlannerListCommandTest.java]: See the above comment.

**Review by chyeo — CHANGES_REQUESTED** [review]

File: src/main/java/pwe/planner/logic/commands/PlannerListCommand.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    public static final String COMMAND_WORD = "planner_list";
+
+    public static final String MESSAGE_SUCCESS = "Listed all degree planners:\n%1$s";
+
+    @Override
+    public CommandResult execute(Model model, CommandHistory history) {
+        requireNonNull(model);
+
+        model.updateFilteredDegreePlannerList(PREDICATE_SHOW_ALL_DEGREE_PLANNERS);
+        String degreePlannerListContent =
```
Comment (chyeo) [inline src/main/java/pwe/planner/logic/commands/PlannerListCommand.java]: the following lines can be compressed into 2 lines.

**Review by chyeo — CHANGES_REQUESTED** [review]

**Comment by chyeo** [comment]
Looks good on my side, will tag @lycjackie to view this PR as he is in charge of `Logic` components.

**Review by chyeo — APPROVED** [review]

**Review by lycjackie — APPROVED** [review]
LGTM

Approve and merging

**Pull request merged**